src/figuresClasses.ts

```diff
@@ -2,10 +1,101 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+function triangleArea(a: number, b: number, c: number): number {
+  const s = (a + b + c) / 2;
+
+  return Math.sqrt(s * (s - a) * (s - b) * (s - c));
+}
+
+function circleArea(radius: number): number {
+  return radius ** 2 * Math.PI;
+}
+
+function rectangleArea(width: number, height: number): number {
+  return width * height;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape = 'triangle',
+  ) {
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to call `this.getArea();` in the constructor.